From 9a5c90d8b8569b40cd9748bcd7f3d0a66d5fced4 Mon Sep 17 00:00:00 2001
From: Rafael Antognolli <rafael.antognolli@intel.com>
Date: Tue, 8 Nov 2016 15:47:59 -0800
Subject: [PATCH 3/3] Fix display null pointer exception on colorModes array.

If HwC doesn't implement or support getColorModes, a null array is going
to be used by updateColorModesLocked inside LocalDisplayAdapter. So we
need to check for that and avoid a null pointer exception.

Bug ID: https://github.com/android-ia/external-drm_hwcomposer/issues/6
Test: Boots up to home screen.
---
 services/core/java/com/android/server/display/LocalDisplayAdapter.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/services/core/java/com/android/server/display/LocalDisplayAdapter.java b/services/core/java/com/android/server/display/LocalDisplayAdapter.java
index 61c2eac..4827b9d 100644
--- a/services/core/java/com/android/server/display/LocalDisplayAdapter.java
+++ b/services/core/java/com/android/server/display/LocalDisplayAdapter.java
@@ -285,6 +285,9 @@ final class LocalDisplayAdapter extends DisplayAdapter {
                 int activeColorMode) {
             List<Integer> pendingColorModes = new ArrayList<>();

+            if (colorModes == null)
+                return false;
+
             // Build an updated list of all existing color modes.
             boolean colorModesAdded = false;
             for (int colorMode: colorModes) {
--
2.7.4
